### ISSUE TITLE
Error when parsing samples with spaces at the beginning

### ISSUE THREAD

**Issue by StudyingFather** [jmerle/competitive-companion#149]
When I parsed [this problem](https://www.luogu.com.cn/problem/P1185), the spaces at the beginning in the sample were missing.

Here is the data:

```json
{
    "name": "P1185 绘制二叉树",
    "group": "Luogu",
    "url": "https://www.luogu.com.cn/problem/P1185",
    "interactive": false,
    "memoryLimit": 128,
    "timeLimit": 1000,
    "tests": [
        {
            "input": "2 0\n",
            "output": "o  \n / \\ \no   o\n"
        },
        {
            "input": "4 0\n",
            "output": "o           \n          / \\          \n         /   \\         \n        /     \\        \n       /       \\       \n      /         \\      \n     o           o     \n    / \\         / \\    \n   /   \\       /   \\   \n  o     o     o     o  \n / \\   / \\   / \\   / \\ \no   o o   o o   o o   o\n"
        },
        {
            "input": "4 3\n3 2\n4 1\n3 4\n",
            "output": "o           \n          / \\          \n         /   \\         \n        /     \\        \n       /       \\       \n      /         \\      \n     o           o     \n    /           /      \n   /           /       \n  o           o        \n   \\         / \\       \n    o       o   o\n"
        }
    ],
    "testType": "single",
    "input": {
        "type": "stdin"
    },
    "output": {
        "type": "stdout"
    },
    "languages": {
        "java": {
            "mainClass": "Main",
            "taskClass": "P1185"
        }
    },
    "batch": {
        "id": "9fa8ca28-f6c4-4e26-a908-a4de1f80ae31",
        "size": 1
    }
}
```

**Comment by ouuan** [comment]
https://github.com/jmerle/competitive-companion/blob/f271e64bbb337ebb11963aa3e34ff11a50de6983/src/parsers/problem/LuoguProblemParser.ts#L24-L25

The test cases are trimmed.

I think it's better to trim blank lines at the end of the document and the trailing spaces at the end of each line, instead of using `String.prototype.trim()`.

**Comment by jmerle** [comment]
Thanks for reporting the issue, fixed in 5dbcf3b63d9ab821175b795ad55a1e6e549a9a87. From the next release onwards all sample input/output will be normalized as per @ouuan's idea, trimming trailing spaces at the end of each line and at the end of the document. Will be released within an hour.
